src/OrchardCore.Modules/OrchardCore.Media/Processing/MediaFileCache.cs

```diff
@@ -0,0 +1,142 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OrchardCore.Environment.Shell;
+using SixLabors.ImageSharp.Web;
+using SixLabors.ImageSharp.Web.Caching;
+using SixLabors.ImageSharp.Web.Helpers;
+using SixLabors.ImageSharp.Web.Middleware;
+using SixLabors.ImageSharp.Web.Resolvers;
+
+namespace OrchardCore.Media.Processing
+{
+    /// <summary>
+    /// A cache for media assets, per tenant.
+    /// </summary>
```

Comment:
Open to suggestions here. This basically makes the cache tenant only in the App_Data folder.

Upside to this is that cached files are not available in the `wwwroot` folder of the `OrchardCore.Cms.Web` startup project, and we can lose all those issues of needing placeholders in the `wwwroot` folder. To me seems strange to store files for internal use in `wwwroot`, but as I said open to options. 
Is possible to extend the `wwwroot` folders to include tenant names so that cache could be cleared on a tenant by tenant basis. But even that maybe is something only the host server / admin should be doing?

Reply:
By using the wwwroot folder we can use the default static file provider, and just change the Path of the request to point to a "cached/preloaded" file. The cache can still be per tenant in a sub folder. The other benefit is that when you backup your App_data, you don't have to back the cache.

I am sure we can have a custom IS cache implementation that would append the tenant in the folder. This can also be done as a separate PR.

It's important to abstract the file cache implementation so that we can provide custom ones for invalidation logic, or custom storage (storing metadata in a sqlite db for instance, or even the content).

Reply:
> By using the wwwroot folder we can use the default static file provider, and just change the Path of the request to point to a "cached/preloaded" file. The cache can still be per tenant in a sub folder. The other benefit is that when you backup your App_data, you don't have to back the cache.

Cool, will move to wwwroot as a default `IFileStoreCache` which wraps the `IFileProvider`

> I am sure we can have a custom IS cache implementation that would append the tenant in the folder. This can also be done as a separate PR.

Yes, that's easy. 

> It's important to abstract the file cache implementation so that we can provide custom ones for invalidation logic, or custom storage (storing metadata in a sqlite db for instance, or even the content).

Cool, I see where you're going with this.